solidity/contracts/middleware/InterchainAccountRouter.sol

```diff
@@ -720,6 +743,59 @@ contract InterchainAccountRouter is Router {
             );
     }
 
+    function callRemoteCommitReveal(
+        uint32 _destination,
+        bytes32 _router,
+        bytes32 _ism,
+        bytes memory _hookMetadata,
+        IPostDispatchHook _hook,
+        bytes32 _salt,
+        bytes32 _commitment
+    ) public payable returns (bytes32 _commitmentMsgId, bytes32 _revealMsgId) {
+        bytes memory _commitmentMsg = InterchainAccountMessage
+            .encodeCommitment({
+                _owner: msg.sender.addressToBytes32(),
+                _ism: _ism,
+                _commitment: _commitment,
+                _userSalt: _salt
+            });
+        bytes memory _revealMsg = InterchainAccountMessage.encodeReveal({
+            _ism: _ism,
```

Comment:
is this the CCIP Read ISM? as written you force the user to use the same for security and liveness